gluten-substrait/src/main/scala/org/apache/gluten/execution/WholeStageTransformer.scala

```diff
@@ -421,17 +420,7 @@ case class WholeStageTransformer(child: SparkPlan, materializeInput: Boolean = f
       )
     )
 
-    allInputPartitions.head.indices.foreach(
-      i => {
-        val currentPartitions = allInputPartitions.map(_(i))
-        currentPartitions.indices.foreach(
-          i =>
-            currentPartitions(i) match {
-              case f: FilePartition =>
-                SoftAffinity.updateFilePartitionLocations(f, rdd.id)
-              case _ =>
-            })
-      })
+    SoftAffinity.updateFilePartitionLocations(Seq(allInputPartitions), rdd.id)
```

Comment:
Since we don't need IndexSeq, could you remove the `.toIndexedSeq` in https://github.com/apache/incubator-gluten/pull/9079/files#diff-5690472af86b8a7836d5b2e167d22ce95953758c29f9889d0a25515fc7f07aa4R385